File: packages/ios/FormbricksSDK/FormbricksSDK/Model/Error/FormbricksAPIError.swift
struct FormbricksAPIError: Codable, LocalizedError {
    let code: String
    let message: String
    let details: [String:String]?
}

extension FormbricksAPIError {
    func getDetailedErrorMessage() -> String {
        if let errorDetails = details?.map({ "\($0.key): \($0.value)" }) {
            return "\(message)\n\(errorDetails)"
        } else {
            return message
        }
    }
}

File: packages/android/formbricksSDK/src/main/java/com/formbricks/formbrickssdk/api/error/FormbricksAPIError.kt
package com.formbricks.formbrickssdk.api.error

import com.google.gson.annotations.SerializedName

data class FormbricksAPIError(
    @SerializedName("code") val code: String,
    @SerializedName("message") val messageText: String,
    @SerializedName("details") val details: Map<String, String>? = null
) : RuntimeException(messageText)